modules/secrets-store-aws-secrets-manager.adoc

```diff
@@ -0,0 +1,332 @@
+          resources:
+            requests:
+              cpu: 50m
+              memory: 100Mi
+            limits:
+              cpu: 50m
+              memory: 100Mi
+          securityContext:
+            privileged: true
+          volumeMounts:
+            - mountPath: "/etc/kubernetes/secrets-store-csi-providers"
+              name: providervol
+            - name: mountpoint-dir
+              mountPath: /var/lib/kubelet/pods
+              mountPropagation: HostToContainer
+      tolerations:
+      - operator: Exists
+      volumes:
+        - name: providervol
+          hostPath:
+            path: "/etc/kubernetes/secrets-store-csi-providers"
+        - name: mountpoint-dir
+          hostPath:
+            path: /var/lib/kubelet/pods
+            type: DirectoryOrCreate
+      nodeSelector:
+        kubernetes.io/os: linux
+----
++
+[NOTE]
+====
+This configuration is modified from the configuration provided in the link:https://github.com/aws/secrets-store-csi-driver-provider-aws#installing-the-aws-provider[AWS documentation] so that it works properly with {product-title}.
+====
+
+.. Grant privileged access to the `csi-secrets-store-provider-aws` service account by running the following command:
++
+[source,terminal]
+----
+$ oc adm policy add-scc-to-user privileged -z csi-secrets-store-provider-aws -n openshift-cluster-csi-drivers
+----
+
+.. Create the provider resources by running the following command:
++
+[source,terminal]
+----
+$ oc apply -f aws-provider.yaml
+----
+
+. Grant permission to allow the service account to read the AWS secret object:
+
+.. Create a directory to contain the credentials request by running the following command:
++
+[source,terminal]
+----
+$ mkdir credentialsrequest-dir-aws
+----
+
+.. Create a YAML file with the following configuration for the credentials request:
++
+.Example `credentialsrequest.yaml` file
+[source,yaml]
+----
+apiVersion: cloudcredential.openshift.io/v1
+kind: CredentialsRequest
+metadata:
+  name: aws-provider-test
+  namespace: openshift-cloud-credential-operator
+spec:
+  providerSpec:
+    apiVersion: cloudcredential.openshift.io/v1
+    kind: AWSProviderSpec
+    statementEntries:
+    - action:
+      - "secretsmanager:GetSecretValue"
+      - "secretsmanager:DescribeSecret"
+      effect: Allow
+      resource: "arn:*:secretsmanager:*:*:secret:testSecret-??????"
+  secretRef:
+    name: aws-creds
+    namespace: test-namespace
+  serviceAccountNames:
+  - aws-provider
+----
++
+// TODO `resource` value: should we say something about this? I changed it from "secret:xxiaSecret" to "secret:testSecret". What are the "??????" supposed to be?
+
+.. Retrieve the OIDC provider by running the following command:
++
+[source,terminal]
+----
+$ oc get --raw=/.well-known/openid-configuration | jq -r '.issuer'
+----
++
+// TODO: Can this be rewritten to use jsonpath or something else instead of jq? (To avoid making users install jq)
```

Comment:
@xingxingxia Is there any way to get the issuer value here without using jq? I ask because we try to avoid using jq in our procedures when possible, to avoid making users have to install jq if they don't already have it. 

When I tried, I saw that we can't use --raw and -o can't be used together. But I wasn't sure if there was any other way. Thoughts?

Reply:
Below are the full output:
```
$ oc get --raw=/.well-known/openid-configuration
{"issuer":"https://xxxx.s3.ap-northeast-2.amazonaws.com","jwks_uri":"https://10.0.60.253:6443/openid/v1/jwks","response_types_supported":["id_token"],"subject_types_supported":["public"],"id_token_signing_alg_values_supported":["RS256"]}
```
`"issuer":"https://xxxxxxxx"` is OIDC URL. Note the whole `xxxxxxxx` after the `https://` is the OIDC provider name (not the OIDC provider arn. ARN has prefixed `arn:aws:.....<aws_account_id>`).

Reply:
Okay thanks. I don't see another easy way to just get the issuer value cleanly, so I will keep jq, and just add it as a prereq, thanks!